### DISCUSSION TITLE
How to use test filters without triggering "obsolete snapshot"

### DISCUSSION THREAD

**Discussion by jasonkuhrt** [vitest-dev/vitest#7106] [Q&A] [answered]
Hey, when running a command like:

```
vitest run cli/schema --testNamePattern 'json: true.*should see Invalid Token error.*'
```

The targeted test runs but then all other tests are considered obsolete.

```
  Snapshots  1 failed
             70 obsolete
```

The major usability problem here is that while filtering I cannot update the failing snapshot without causing the others to be deleted, but if I don't filter I have many more tests that are slow etc.

Am I missing something?

**Comment by hi-ogawa** [comment]
Can you provide a repro? I don't think this should happen with a simple `toMatchSnapshot()` API. Do you happen to use explicit snapshot name like `toMatchSnapshot("my-name")`?

**Reply by jasonkuhrt** [reply]
Yeah we are using explicit snapshot names.

The repo is open source so I can share that later today when I start work.

**Reply by jasonkuhrt** [reply]
Once this PR is merged I will give you a set of simple repro steps. https://github.com/graphql-hive/console/actions/runs/12425451027/job/34692481602.

**Reply by hi-ogawa** [reply, marked as answer]
~I thought this is a limitation since we cannot track what's the snapshot until `toMatchSnapshot("my-name")`, but I think we can probably apply a same heuristics as how we handle `toMatchSnapshot()` which uses auto-icremented number.~
(EDIT: oh wait, `toMatchSnapshot()` actually has the same behavior... :cry:)

https://stackblitz.com/edit/vitest-dev-vitest-mitcgkkj?file=test%2F__snapshots__%2Fbasic.test.ts.snap

```js
import { expect, test } from 'vitest';

test('x test', () => {
  expect(0).toMatchSnapshot();  // generates `x test 1`
  expect(0).toMatchSnapshot();  // generates `x test 2`  // when first one fails `x test 2` is considered obsolete
});

test('y test', () => {
  expect(0).toMatchSnapshot('foo'); // generates `y test > bar 1`
  expect(0).toMatchSnapshot('bar'); // generates `y test > foo 1`  // when first on fails `y test > foo 1` is considered obsolete
});
```

**Reply by hi-ogawa** [reply]
@jasonkuhrt I made an issue https://github.com/vitest-dev/vitest/issues/7113. Can you confirm that's the issue you're seeing?

**Reply by jasonkuhrt** [reply]
@hi-ogawa that looks right to me! Thanks!
